oterm: 0.7.3 -> 0.8.3

gaelj opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

SuperSandro2000 commented on pkgs/development/python-modules/mcp/default.nix:
```diff
@@ -0,0 +1,54 @@
+
+  pythonImportsCheck = [ "mcp" ];
+
+  # no tests implemented
```
What is this https://github.com/modelcontextprotocol/python-sdk/tree/main/tests then?

gaelj replied:
Thanks for pointing this out ; i fixed this oversight

SuperSandro2000 commented on pkgs/development/python-modules/pydantic-settings/default.nix:
```diff
@@ -14,7 +14,7 @@
   self = buildPythonPackage rec {
     pname = "pydantic-settings";
-    version = "2.5.2";
+    version = "2.7.1";
```
Is this absolutely necessary? Because that must go to staging and probably causes regressions

gaelj replied:
Yes, MCP needed this version to build unfortunately.

FliegendeWurst replied:
Try `pythonRelaxDeps = [ "pydantic-settings" ]` instead.
Upstream also relaxed the requirement https://github.com/modelcontextprotocol/python-sdk/commit/0d48a4494fc60d350ed1b10a88edc8b06f716d65

gaelj replied:
Nice, it works !

FliegendeWurst commented on pkgs/development/python-modules/mcp/default.nix:
```diff
@@ -0,0 +1,53 @@
+  ];
+
+  pythonImportsCheck = [ "mcp" ];
+
```
```suggestion
  nativeCheckInputs = [ pytestCheckHook ];
```
To run the tests.

gaelj replied:
Now that tests are enabled, they are failing. I have successfully solved several errors by adding the following:

```nix
  nativeCheckInputs = [
    pytestCheckHook
    typer
    uv
  ];

  preCheck = ''
    substituteInPlace tests/client/test_stdio.py --replace "/usr/bin/tee" "tee"
  '';

  preBuild = ''
    export HOME=$(pwd)
  '';
```

But now I'm stuck with this error, which I cannot find how to solve:

```
nix build .#python3Packages.mcp
warning: Git tree '/home/gaj/github.com/gaelj/nixpkgs' is dirty
error: builder for '/nix/store/32p2fgxh1gb6yr4ih5v46qwxchvcz3k7-python3.12-mcp-1.2.0.drv' failed with exit code 1;
       last 25 log lines:
       >             server_name=server_name,
       >         )
       >         assert success
       >     
       >         # Read the generated config
       >         config_file = mock_config_path / "claude_desktop_config.json"
       >         config = json.loads(config_file.read_text())
       >     
       >         # Get the command and args
       >         server_config = config["mcpServers"][server_name]
       >         command = server_config["command"]
       >         args = server_config["args"]
       >     
       >         test_args = [command] + args + ["--help"]
       >     
       >         result = subprocess.run(test_args, capture_output=True, text=True, timeout=5)
       >     
       > >       assert result.returncode == 0
       > E       AssertionError: assert 2 == 0
       > E        +  where 2 = CompletedProcess(args=['uv', 'run', '--with', 'mcp', 'mcp', 'run', '/build/mcp-1.2.0/test_server.py:app', '--help'], r...r='error: Could not read ELF interpreter from any of the following paths: /bin/sh, /usr/bin/env, /bin/dash, /bin/ls\n').returncode
       >
       > tests/client/test_config.py:51: AssertionError
       > =========================== short test summary info ============================
       > FAILED tests/client/test_config.py::test_command_execution - AssertionError: assert 2 == 0
       > ======================== 1 failed, 141 passed in 1.05s =========================
       For full logs, run 'nix log /nix/store/32p2fgxh1gb6yr4ih5v46qwxchvcz3k7-python3.12-mcp-1.2.0.drv'.
```

Any help/tips would be much appreciated !

SuperSandro2000 replied:
The uv somehow fails to find its libraries. Is it unpatched? some precompiled binary? Worst case just skip the test.

gaelj replied:
It's a [Python wrapper](https://github.com/NixOS/nixpkgs/blob/b517231a300860a6bd8fa4aacb3ebf9d30d4e383/pkgs/development/python-modules/uv/default.nix#L10) around a [Rust package](https://github.com/NixOS/nixpkgs/blob/b517231a300860a6bd8fa4aacb3ebf9d30d4e383/pkgs/by-name/uv/uv/package.nix#L19).

Anyway, inspecting these packages didn't ring any bells so I guess I'll remove the tests.

gaelj commented:
In the meantime, someone else added `mcp`, so I removed it from here and rebased on latest master

gaelj committed:
maintainers: add gaelj

gaelj committed:
oterm: assign gaelj as maintainer

gaelj committed:
textualeffects: init at 0.1.3

textualeffects: integrate code-review comments

textualeffects: integrate code-review comments

gaelj committed:
oterm: fix runtime error: missing textualeffects (#366021)

gaelj committed:
oterm: 0.8.1 -> 0.8.3

gaelj committed:
oterm: 0.7.3 -> 0.8.0

gaelj committed:
oterm: 0.8.0 -> 0.8.1
